content/reference/forms.md

```diff
@@ -0,0 +1,311 @@
+While paragraph text is generally straight forward to place on a page, lists &
+forms require some more work. This section explains everything you need to know
+to work with forms in Choo.
+
+This guide assumes you're working from a project generated by `create-choo-app`.
+This allows us to write inline CSS, which makes examples a little simpler to
+read. However our goal is to provide you with knowledge that translates to any
+setup - even if you don't end up using Choo.
+
+## How do forms work?
+Before we dive into how forms work in Choo, let's dive into how forms work on
+pages that don't use any JavaScript. A lot of the web was designed to work
+without JavaScript, and to submit forms, you don't need any JS at all. Knowing
+what the default behavior is of forms allows us to build on top of it, rather
+than trying to rewrite functionality that's already available to us.
+
+### The Form Element
+Forms are declared using the form tag. By themselves they don't do much, but
+they have a few important attributes that are good to know about.
+
+The first attribute is `method=""`. This tells the form which HTTP method to
+use. By default it's set to `POST`, so often it's not needed to define this.
+
+The second attribute is `action=""`. This attribute tells the form where to
+redirect the page to when the submission was successful.
+
+Forms also always should have an `id=""` attribute on them. This makes them
+easier to debug, and shows up in the payload that's sent to the server.
+
+```html
+<form id="login" action="/dashboard">
+</form>
+```
+
+### The Input Element
+Forms need data. And `<input>` elements provide that data. As a rule, each
+`<input>` element has a `type=""` attribute, and an accompanying `<label>`
+element. They also need a `name=""` and an `id=""`. That's quite a bit of data
+required. But together it allows you to create a wide range of input.
+
+```html
+<form id="login" action="/dashboard">
+  <label for="username">username</label>
+  <input id="username" name="username" type="text">
+  <label for="password">password</label>
+  <input id="password" name="password" type="password">
+  <input type="submit" value="Login">
+</form>
+```
+
+### Validating Input
+Forms come with a wide range of validation built in. Probably the biggest
+benefit is that it works on all platforms, with little effort.  It respects user
+settings such as font-size, and supports screen readers out of the box.
+
+To validate the form's input fields, there's a few attributes we can use:
+- `pattern` - validate the form's input field using a Regular Expression. For
+  example `pattern="^.{1,15}$"` makes sure strings have a length of at least 1,
+  and not more than 15.
+- `required` - make sure that the field is filled in, and valid.
+- `title` - the message to display if the `pattern` attribute is invalid. This
+  is useful for users that can't read RegExes in their error messages (e.g.
+  everyone).
```

Comment:
if this is everyone, then the sentence itself is kind of redundant

Reply:
Hehe, yeah this was more of a joke than anything else. But it looks like it doesn't always land, so it's okay to remove.